cfme/tests/infrastructure/test_timelines.py

```diff
@@ -30,213 +25,161 @@ def a_provider(request):
-    server_name = "{name} ({sid})".format(name=appliance.server.name, sid=appliance.server.sid)
-    relations_view.form.server.select_by_visible_text(server_name)
-    relations_view.form.save_button.click()
-    yield
-    # unset diagnostics vm
-    relations_view = navigate_to(new_vm, 'EditManagementEngineRelationship')
-    relations_view.form.server.select_by_visible_text('<Not a Server>')
-    relations_view.form.save_button.click()
-
-
-@pytest.fixture(scope="module")
-def gen_events(new_vm):
-    logger.debug('Starting, stopping VM')
-    mgmt = new_vm.provider.mgmt
-    mgmt.stop_vm(new_vm.name)
-    mgmt.start_vm(new_vm.name)
-
-
-def count_events(target, vm):
-    timelines_view = navigate_to(target, 'Timelines')
-    if isinstance(target, Server):
-        timelines_view = timelines_view.timelines
-    timelines_view.filter.time_position.select_by_visible_text('centered')
-    timelines_view.filter.apply.click()
-    found_events = []
-    for evt in timelines_view.chart.get_events():
-        if not hasattr(evt, 'source_vm'):
-            # BZ(1428797)
-            logger.warn("event {evt} doesn't have source_vm field. Probably issue".format(evt=evt))
-            continue
-        elif evt.source_vm == vm.name:
-            found_events.append(evt)
-
-    logger.info("found events: {evt}".format(evt="\n".join([repr(e) for e in found_events])))
-    return len(found_events)
-
-
-def test_infra_provider_event(gen_events, new_vm):
-    """Tests provider event on timelines
-
-    Metadata:
-        test_flag: timelines, provision
-    """
-
-    wait_for(count_events, [new_vm.provider, new_vm], timeout='5m', fail_condition=0,
-             message="events to appear")
+    vm = VM.factory(random_vm_name('timelines', max_length=16), a_provider)
+    logger.info('Fixture new_vm set up! Name: %r Provider: %r', vm.name, vm.provider.name)
+    yield vm
+    logger.info('Fixture new_vm teardown! Name: %r Provider: %r', vm.name, vm.provider.name)
 
 
-def test_infra_host_event(appliance, a_provider, gen_events, new_vm):
-    """Tests host event on timelines
+class VMEvent(object):
+    """Class for generating  events on a VM in order to check it on Timelines. According to the
+    event passed in the parameter it will perform the action on the given VM.
 
-    Metadata:
-        test_flag: timelines, provision
+    Args:
+        vm: A VM object (Object)
+        event: An event which will be generated (string).
     """
-    view = navigate_to(new_vm, "Details")
-    host_name = view.entities.relationships.get_text_of('Host')
-    host_collection = appliance.collections.hosts
-    host = host_collection.instantiate(name=host_name, provider=a_provider)
-    wait_for(count_events, [host, new_vm], timeout='10m', fail_condition=0,
-             message="events to appear")
 
+    def __init__(self, vm, event):
+        self.vm = vm
+        self.action = event
+
+        if self.action == 'create':
```

Comment:
REQUIRED: there is advice from @psav to replace these ifs with dict and I agree with him. Please move all these statements to dict